server/src/main/java/org/elasticsearch/http/HttpServerTransport.java

```diff
@@ -52,5 +54,8 @@ interface Dispatcher {
          */
         void dispatchBadRequest(RestChannel channel, ThreadContext threadContext, Throwable cause);
 
+        default Map<String, HttpRouteStats> getStats() {
```

Comment:
NIT: I think it is worth renaming to `routeStats` or `perRouteStats` to be consistent with `stats` method. That one could also have some reasonable default now.